src/hooks/useFormData.ts

```diff
@@ -0,0 +1,71 @@
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { useState } from 'react';
+import { ERROR_MESSAGES } from '../messages';
+
+export type FetchedAddress = {
+  prefecture: string;
+  area: string;
+};
+
+export type AddressForm = FetchedAddress & { postalCode: string };
+
+export const useFormData = () => {
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    formState: { errors },
+  } = useForm<AddressForm>({
+    mode: 'onBlur',
+    defaultValues: {
+      postalCode: '',
+      prefecture: '',
+      area: '',
+    },
+  });
+
+  const [noResultsMessage, setNoResultsMessage] = useState<string>('');
+
+  const onSubmit: SubmitHandler<AddressForm> = (data) => console.log(data);
+  const fetchData = async (query: string) => {
+    try {
+      const response = await fetch(`https://zipcloud.ibsnet.co.jp/api/search?zipcode=${query}`);
+      const result = await response.json();
+      if (result['results']) {
+        const prefecture = result['results'][0]['address1'];
+        const area = result['results'][0]['address2'];
+        const fetchedAddress: FetchedAddress = { prefecture: prefecture, area: area };
+        return fetchedAddress;
+      } else {
+        const emptyAddress: FetchedAddress = { prefecture: '', area: '' };
+        return emptyAddress;
+      }
+    } catch (error) {
+      console.error('Error fetching data:', error);
+      const emptyAddress: FetchedAddress = { prefecture: '', area: '' };
+      return emptyAddress;
```

Comment:
Instead of constructing empty objects and returning it out, you might want to use `throw error` instead to trigger `catch` block in line 63 and handle the error there. Without this, that catch block might never be executed.

```suggestion
      throw error
```

Reply:
Added error throw